src/components/Journal/JournalEntryForm/TagDimensionsGroup.tsx

```diff
@@ -24,7 +24,7 @@ export const TagDimensionsGroup = ({
 
   const handleTagValueChange = (dimensionKey: string) => (newTag: Tag | null) => {
     const filteredTags = value.filter(tag =>
-      tag.dimensionLabel.toLowerCase() !== dimensionKey.toLowerCase(),
+      tag.key.toLowerCase() !== dimensionKey.toLowerCase(),
```

Comment:
Recognizing that `TagDimensionGroup` is currently nested under JournalEntryForm and shouldn't be, I'll move it.